gradle-plugin/src/main/java/com/dpforge/manifestguard/ManifestGuardPlugin.kt

```diff
@@ -15,7 +15,7 @@ class ManifestGuardPlugin : Plugin<Project> {
         project.plugins.withType(AppPlugin::class.java) {
 
             val androidComponents =
-                project.extensions.getByType(AndroidComponentsExtension::class.java)
+                project.extensions.getByType(ApplicationAndroidComponentsExtension::class.java)
```

Comment:
more direct extension, but AndroidComponentsExtension will work here too, because it is inherited https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-api/src/main/java/com/android/build/api/variant/ApplicationAndroidComponentsExtension.kt?q=ApplicationAndroidComponentsExtension